ragaai_catalyst/synthetic_data_generation.py

```
@@ -327,6 +326,59 @@ def _generate_llm_response(self, text, system_message, model_config, api_key=Non
 
         json_data = json.loads(content)
         return pd.DataFrame(json_data)
+    
+    def _generate_raw_llm_response(self, text, system_message: Optional[str] = None, model_config: Dict[str, Any] = dict(), api_key=None):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)